WordPress/Classes/Stores/NoticeStore.swift

```diff
@@ -62,6 +65,10 @@ struct Notice {
         self.actionHandler = actionHandler
         self.style = style
     }
+
+    static func ==(lhs: Notice, rhs: Notice) -> Bool {
```

Comment:
I thought about using full `struct` equality but that turned into a mess because then I would need to implement `Equatable` for all the other structs used in `Notice`. There's also the `Dictionary` properties with `Any` in them. It became very complex so I abandoned it and used `identifier` instead.

Reply:
Couldn't we just rely on the `title` here then? Given that those values here are basically pseudo-randomly generated, this feels like a hacky reimplementation of `===`?

Reply:
wait those are structs so `===` doesn't apply, nevermind me. 

I'm confused how can this comparison result ever return `true` though then? Wouldn't this basically require a UUID collision?

Reply:
@jklausa 

Yeah, I guess this can be confusing. The reason why I needed `Equatable` was to implement `NoticeStore.clearNotice` where we use equality to find the `Notice` to remove. 

```swift
pending.removeAll { $0 == notice }
```

The `clearNotice`, executed by `NoticeAction.clear`, is used in `NoticePresenter` where we _keep_ a reference to the `Notice`:

```swift
private func presentNoticeInBackground(_ notice: Notice) -> NoticeArtifact? {

       let dismiss = {
            ActionDispatcher.dispatch(NoticeAction.clear(notice))
        }

       // Later in the method:
       DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + Animations.dismissDelay, execute: dismiss)
}
```

The `Equatable` implementation here relies on the `Notice` struct (and its properties) to be **immutable**. When a `Notice` is initialized, it will have a unique `identifier` property value. And we use that in `==` to compare when the code above is executed. 

So you're right that it _is_ kind of a hack-ish. Now I realize that `==` is **both** a value and identity comparison. I'm open to suggestions though!

Reply:
Does `identifier` make `tag` redundant? Can the network errors be dismissed the same way (keeping a copy of the notice)? Or maybe quick start should change to using tags?

Reply:
@nheagy I don't think so. I foresee the `tag` (`clearWithTag` Action) to be used more since it's going to be cumbersome to keep the `Notice` instances in the code. I created `clear(Notice)`, which uses the `identifier`, to get around the possibility of dismissing the wrong (current) `Notice` in the `DispatchQueue.main.asyncAfter` handler.

This is what can happen if we use `NoticeAction.dismiss` instead:

![Screen Recording 2019-03-15 at 6 05 00 PM 2019-03-15 18_09_23](https://user-images.githubusercontent.com/198826/54467885-ae124e80-474d-11e9-9a1a-196698652f82.gif)

Sorry for the bad quality. 😅

Reply:
I wasn't suggesting using `.dismiss` instead, but that both could use the same method, either `clear(Notice)` or `clearWithTag()`. Now that `clearWithTag()` exists, that may be the best method for Quick Start to use as well.

Reply:
Oh, yes of course. I was just trying to explain why `clear(Notice)` was created in the first place. If it wasn't for the race condition, it wouldn't have existed since `clearWithTag()` is easier to use.